exercicios/para-casa/Sabrina-Lopes/entrega.js

```diff
@@ -0,0 +1,97 @@
+//Gatos
+
+
+function Gato(nome, idade, cor, castrado){
+    this.nome = nome;
+    this.idade = idade;
+    this.cor = cor;
+    this.castrado = castrado;
+    this.historico = new HistoricoMedico()
+
+}
+
+Gato.prototype.miar = function(){
+    console.log(`0 gato ${this.nome} está miando`)
+}
+//cachorros
+
+
+
+function Cachorro(nome, idade, cor, castrado, raca){
+    this.nome = nome;
+    this.idade = idade;
+    this.cor = cor;
+    this.castrado = castrado;
+    this.raca = raca;
+    this.historico = new HistoricoMedico();
+}
+
+Cachorro.prototype.latir = function(){
+    console.log(`O cachorro ${this.nome} está latindo`)
+}
+
+//Animais Exóticos
+
+
+
+function AnimalExotico(nome, idade, cor, especie){
+    this.nome = nome;
+    this.idade = idade;
+    this.cor =  cor;
+    this.especie = especie;
+    this.historico = new HistoricoMedico();
+    
+}
+
+    AnimalExotico.prototype.ehAdulto = function(){
+        if(this.especie === 'hamster' && this.idade >=2){
+            return true;
+        } else if(this.especie ==='papagaio' && this.idade >=36){
+            return true;
+        }else if (this.especie === 'calopsita' && this.idade >=18){
+            return true;
+        }else{
+            return false;
+        }
+    }
+
+//Histórico Médico
+
+function HistoricoMedico(){
+    this.consultas = [];
+    this.vacinas = [];
+    this.procedimentos = [];
+}
+
+HistoricoMedico.prototype.registrarConsulta = function(data){
+    this.consultas.push(data);
+    console.log(`Consulta registrada em ${data}`);
+}
+
+HistoricoMedico.prototype.aplicarVacina = function(vacina){
+    this.vacinas.push(vacina);
+    console.log(`Vacina ${vacina} aplicada`);
+}
+
+HistoricoMedico.prototype.registrarProcedimento = function(procedimento){
```

Comment:
uma outra sugestão é incluir na função do registro do procedimento o motivo do mesmo, eu fiquei com um pouco de dúvida nessa parte, mas vou deixar aqui embaixo como eu tentei fazer. 

this.realizarProcedimento = function realizarProcedimento(tipo, motivo) {
    let procedimento = {
      tipo: tipo,
      motivo: motivo,
    };
    this.procedimentos.push(procedimento);
    console.log(`o procedimento ${tipo} foi realizado`);
  };